### PR TITLE
Use 'hatch version' instead of manually updating the version

### PR DESCRIPTION
**Pull request by felixfontein**
We did this change to antsibull-docs some time ago. I had to fix that a bit today with https://github.com/ansible-community/antsibull-docs/commit/676780b99912f5e9d8b206b14a65ed03d8405919. This PR applies the noxfile changes to this project as well.

### CODE REVIEW

**Comment by felixfontein** [comment]
Restart CI.

**Review by gotmax23 — CHANGES_REQUESTED** [review]
Make sure you switch the `git add` commands to add the `__init__.py` file instead of `pyproject.toml` after bumping the version.

File: noxfile.py [hunk 1]
```diff
@@ -253,7 +244,13 @@ def bump(session: nox.Session):
...
-        session.run("git", "add", "pyproject.toml", fragment_file, external=True)
+        session.run(
+            "git",
+            "add",
+            "src/antsibull_changelog/__init__.py",
+            fragment_file,
+            external=True,
+        )
         session.run("git", "commit", "-m", f"Prepare {version}.", external=True)
     session.run("antsibull-changelog", "release")
```
Comment (gotmax23) [inline noxfile.py:255]:
```
nox > antsibull-changelog release
ERROR You need to explicitly specify the version for other projects with --version
nox > Command antsibull-changelog release failed with exit code 5
nox > Session bump failed.
```

```suggestion
    session.run("antsibull-changelog", "release", "--version", version)
```

Comment (felixfontein) [reply noxfile.py:255]: Oh right, antsibull-changelog doesn't do magic :facepalm:

**Review by gotmax23 — APPROVED** [review]

File: noxfile.py (hunk 1 above)
Comment (gotmax23) [reply noxfile.py]: See https://github.com/ansible-community/antsibull-changelog/pull/141 :)

**Comment by felixfontein** [comment]
@gotmax23 thanks for reviewing and testing!

**Pull request merged**